sdks/go/test/integration/io/xlang/bigquery/bigquery.go

```diff
@@ -0,0 +1,186 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package bigquery contains integration tests for cross-language BigQuery IO transforms.
+package bigquery
+
+import (
+	"fmt"
+	"math/rand"
+	"reflect"
+	"strings"
+
+	"github.com/apache/beam/sdks/v2/go/pkg/beam"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/io/xlang/bigqueryio"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/register"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/testing/passert"
+)
+
+func init() {
+	register.DoFn2x0[[]byte, func(TestRow)](&CreateTestRowsFn{})
+	register.Emitter1[TestRow]()
+	//register.Function1x1[TestRowPtrs, TestRow](castFn)
+	beam.RegisterType(reflect.TypeOf((*TestRow)(nil)))
+	beam.RegisterType(reflect.TypeOf((*RandData)(nil)))
+	beam.RegisterType(reflect.TypeOf((*TestRowPtrs)(nil)))
+	beam.RegisterType(reflect.TypeOf((*RandDataPtrs)(nil)))
```

Comment:
With the generic registrations, it should be possible to avoid registering these explicitly.

Reply:
Done.